stdlib/rubygems/0/version.rbs

```diff
@@ -1,3 +1,228 @@
+  #     generate builds all day long, but as soon as they make a public release,
+  #     the version must be updated.
+  #
+  #
+  # ### Examples
+  #
+  # Let's work through a project lifecycle using our Stack example from above.
+  #
+  # Version 0.0.1
+  # :   The initial Stack class is release.
+  # Version 0.0.2
+  # :   Switched to a linked=list implementation because it is cooler.
+  # Version 0.1.0
+  # :   Added a `depth` method.
+  # Version 1.0.0
+  # :   Added `top` and made `pop` return nil (`pop` used to return the  old top
+  #     item).
+  # Version 1.1.0
+  # :   `push` now returns the value pushed (it used it return nil).
+  # Version 1.1.1
+  # :   Fixed a bug in the linked list implementation.
+  # Version 1.1.2
+  # :   Fixed a bug introduced in the last fix.
+  #
+  #
+  # Client A needs a stack with basic push/pop capability.  They write to the
+  # original interface (no `top`), so their version constraint looks like:
+  #
+  #     gem 'stack', '>= 0.0'
+  #
+  # Essentially, any version is OK with Client A.  An incompatible change to the
+  # library will cause them grief, but they are willing to take the chance (we
+  # call Client A optimistic).
+  #
+  # Client B is just like Client A except for two things: (1) They use the `depth`
+  # method and (2) they are worried about future incompatibilities, so they write
+  # their version constraint like this:
+  #
+  #     gem 'stack', '~> 0.1'
+  #
+  # The `depth` method was introduced in version 0.1.0, so that version or
+  # anything later is fine, as long as the version stays below version 1.0 where
+  # incompatibilities are introduced.  We call Client B pessimistic because they
+  # are worried about incompatible future changes (it is OK to be pessimistic!).
+  #
+  # ## Preventing Version Catastrophe:
+  #
+  # From: http://blog.zenspider.com/2008/10/rubygems-howto-preventing-cata.html
+  #
+  # Let's say you're depending on the fnord gem version 2.y.z. If you specify your
+  # dependency as ">= 2.0.0" then, you're good, right? What happens if fnord 3.0
+  # comes out and it isn't backwards compatible with 2.y.z? Your stuff will break
+  # as a result of using ">=". The better route is to specify your dependency with
+  # an "approximate" version specifier ("~>"). They're a tad confusing, so here is
+  # how the dependency specifiers work:
+  #
+  #     Specification From  ... To (exclusive)
+  #     ">= 3.0"      3.0   ... &infin;
+  #     "~> 3.0"      3.0   ... 4.0
+  #     "~> 3.0.0"    3.0.0 ... 3.1
+  #     "~> 3.5"      3.5   ... 4.0
+  #     "~> 3.5.0"    3.5.0 ... 3.6
+  #     "~> 3"        3.0   ... 4.0
+  #
+  # For the last example, single-digit versions are automatically extended with a
+  # zero to give a sensible result.
+  #
+  class Version
+    include Comparable
+
+    # True if the `version` string matches RubyGems' requirements.
+    #
+    def self.correct?: (_ToS version) -> bool
```

Comment:
[note] It seems that `nil` should not be received because of the deprecation below:

```console
$ ruby -e 'Gem::Version.correct?(nil)'
nil versions are discouraged and will be deprecated in Rubygems 4
```

Reply:
Ah! `_ToS` cannot reject `nil`...

Reply:
Oh, I missed it… 😓 

I want to improve this, but can we do it? (If we can, I will create a new PR)
It's not so bad as-is, though. 😅